### PR TITLE
Added an output channel for the gzi index

### PR DESCRIPTION
**Pull request by muffato**
When bgzip is given the `-i` option, it creates an index (in gzi format). The index needs to be considered an output so that it can be published. This PR adds an optional output channel to capture the index.

The `-I` option is automatically injected to name the index according to the convention.

## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/tabix/bgzip/main.nf [hunk 1]
```diff
@@ -23,8 +24,10 @@ process TABIX_BGZIP {
     in_bgzip = input.toString().endsWith(".gz")
     command1 = in_bgzip ? '-d' : '-c'
     command2 = in_bgzip ? ''   : " > ${prefix}.${input.getExtension()}.gz"
+    gzi_args = args.matches("(^| )-i\\b") ? "-I index/${prefix}.${input.getExtension()}.gz.gzi" : ''
     """
-    bgzip $command1 $args -@${task.cpus} $input $command2
+    mkdir -p index
```
Comment (FriederikeHanssen) [inline modules/tabix/bgzip/main.nf:29]: why does it need its own subfolder?

Comment (muffato) [reply modules/tabix/bgzip/main.nf:29]: Let me try again, but I think I tried first keeping everything in the same folder, and somehow the index was picked up by the "output" channel, not the "gzi" channel

Comment (FriederikeHanssen) [reply modules/tabix/bgzip/main.nf:29]: ah yes I can see why , the ${prefix}* will pick up everything. there is probably a regex way around it.

Comment (muffato) [reply modules/tabix/bgzip/main.nf]: It works ! Thanks for the poke

File: modules/tabix/bgzip/main.nf [hunk 2]
```diff
@@ -23,8 +24,9 @@ process TABIX_BGZIP {
     in_bgzip = input.toString().endsWith(".gz")
     command1 = in_bgzip ? '-d' : '-c'
     command2 = in_bgzip ? ''   : " > ${prefix}.${input.getExtension()}.gz"
+    gzi_args = args.matches("(^| )-i\\b") ? "-I ${prefix}.${input.getExtension()}.gz.gzi" : ''
     """
-    bgzip $command1 $args -@${task.cpus} $input $command2
+    bgzip $command1 $args $gzi_args -@${task.cpus} $input $command2
```
Comment (maxulysse) [inline modules/tabix/bgzip/main.nf:29]: I'd rather we use $args2 instead of $gzi_args as we do in other modules

Comment (FriederikeHanssen) [reply modules/tabix/bgzip/main.nf:29]: why do we need the second one in the first place? Can you not check the $args string for what you need?

**Comment by FriederikeHanssen** [comment]
@muffato sorry for the review delay :/ I was OOO

File: modules/tabix/bgzip/main.nf (hunk 2 above)
Comment (muffato) [reply modules/tabix/bgzip/main.nf]: Good point @FriederikeHanssen : before adding `-I (...)` I should actually double-check it's not already there. And I should check both the short and long forms of the options.

Therefore, the feature becomes two-fold:

1. Capture the `.gzi` file if indexing has been requested (`-i` or `--index`)
2. Name the `.gzi` file according to the standard prefix unless a specific name has been requested (`-I` or `--index-name`)

I'm not keen on using `$args2` because the convention seems to be 1) it refers to the _second_ command of the script, whereas here it would be used in the _first_ command, and 2) it's always initialised from user input (`task.ext.args2`), whereas here we want to complement the existing arguments from `task.ext.args`.
With the latest version, I modify `$args` directly.

**Review by FriederikeHanssen — APPROVED** [review]
🚀 thank you for adding that

**Comment by muffato** [comment]
Thank you very much !
